File: remove_bg.py
import sys
import subprocess

try:
    from PIL import Image
except ImportError:
    subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'Pillow'])
    from PIL import Image

def process_image(input_path, output_path):
    img = Image.open(input_path).convert("RGB")
    data = img.getdata()
    
    new_data = []
    for r, g, b in data:
        brightness = max(r, g, b)
        alpha = brightness
        if alpha > 0:
            a_norm = alpha / 255.0
            new_r = min(255, int(r / a_norm))
            new_g = min(255, int(g / a_norm))
            new_b = min(255, int(b / a_norm))
            new_data.append((new_r, new_g, new_b, alpha))
        else:
            new_data.append((0, 0, 0, 0))
            
    img = Image.new("RGBA", img.size)
    img.putdata(new_data)
    img.save(output_path, "PNG")

# Do the processing
process_image("public/hero-brain.png", "public/hero-brain.png")
print("Successfully removed background!")

File: remove_bg.js
const fs = require('fs');
const path = require('path');

async function removeBackground() {
  try {
    const sharp = require('sharp');
    const imagePath = path.join(__dirname, 'public', 'hero-brain.png');
    const image = sharp(imagePath);
    
    // Convert to raw pixel data
    const { data, info } = await image.raw().ensureAlpha().toBuffer({ resolveWithObject: true });
    
    for (let i = 0; i < data.length; i += info.channels) {
      const r = data[i];
      const g = data[i + 1];
      const b = data[i + 2];
      
      const brightness = Math.max(r, g, b);
      const alpha = brightness;
      
      if (alpha > 0) {
        const aNorm = alpha / 255.0;
        data[i] = Math.min(255, Math.floor(r / aNorm));
        data[i + 1] = Math.min(255, Math.floor(g / aNorm));
        data[i + 2] = Math.min(255, Math.floor(b / aNorm));
        data[i + 3] = alpha;
      } else {
        data[i + 3] = 0; // Fully transparent map
      }
    }
    
    await sharp(data, {
      raw: {
        width: info.width,
        height: info.height,
        channels: info.channels
      }
    })
    .png()
    .toFile(imagePath);
    
    console.log("Successfully processed the image with sharp!");
  } catch(e) {
    console.error("Error with sharp:", e.message);
  }
}

removeBackground();
